File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {

    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    };

    this.findPage=function (page,size) {
        return $http.get("../brand/findPage.do?page="+page+"&size="+size);
    };

    this.findOne=function (id) {
        return $http.post("../brand/findOne.do?id="+id);
    };

    this.add=function (entity) {
        return $http.post("../brand/save.do",entity);
    };

    this.update=function (entity) {
        return $http.post("../brand/update.do",entity);
    };

    this.del = function (selectList) {
        return $http.post("../brand/delete.do?ids="+selectList);
    };

    this.searchEntity = function (page,size,searchEntity) {
        return $http.post("../brand/search.do?page="+page+"&size="+size,searchEntity);
    };

    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do");
    }
});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 查询所有
     * @return
     */
    List<TbBrand> findAll();

    /**
     * 分页查询
     * @param pageNum 当前页码
     * @param pageSize  每页显示数量
     * @return
     */
    PageResult findPage(int pageNum , int pageSize);

    /**
     * 添加
     * @param brand
     */
    void add(TbBrand brand);

    /**
     * 根据id查询一个
     * @param id
     * @return
     */
    TbBrand findOne(Long id);

    /**
     * 修改
     * @param brand
     */
    void update(TbBrand brand);

    /**
     * 删除
     * @param ids
     */
    void delete(long[] ids);

    /**
     * 条件查询+分页
     * @param brand
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPage(TbBrand brand,int pageNum , int pageSize);


    List<Map> selectOptionList();
}
